core/xmake.lua

```diff
@@ -50,8 +50,8 @@ end
 
 -- for mode coverage
 if is_mode("coverage") then
-    add_cxflags("-coverage")
-    add_ldflags("-coverage")
+    add_cxflags("-coverage", "-fprofile-arcs", "-ftest-coverage")
+    add_ldflags("-coverage", "-fprofile-arcs", "-ftest-coverage")
```

Comment:
You can see [travis-ci](https://travis-ci.org/tboox/xmake/jobs/232260363). 

And the following path rules must be guaranteed for source code and "*.gcno" for old gcc version.

```bash
├── src
│   ├── main.c
│   └── main.gcno
```